src/overlays/actors/ovl_En_Sw/z_en_sw.c

```diff
@@ -35,95 +33,1003 @@ const ActorInit En_Sw_InitVars = {
+    Camera* activeCam;
+    f32 lastFrame;
+
+    lastFrame = Animation_GetLastFrame(&D_06000304.common);
+    if (DECR(this->unk_388) != 0) {
+        Math_SmoothStepToF(&this->skelAnime.playSpeed, 0.0f, 0.6f, 1000.0f, 0.01f);
+        return 0;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/func_80B0C020.s")
+    Math_SmoothStepToF(&this->skelAnime.playSpeed, arg1, 0.6f, 1000.0f, 0.01f);
+    if (arg3 == 1) {
+        if (lastFrame < (this->skelAnime.curFrame + this->skelAnime.playSpeed)) {
+            return 0;
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/func_80B0C0CC.s")
+    activeCam = ACTIVE_CAM;
+    if (Math_Vec3f_DistXYZ(&this->actor.world.pos, &activeCam->eye) < 380.0f) {
+        if (DECR(this->unk_440) == 0) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_STALWALL_ROLL);
+            this->unk_440 = 4;
+        }
+    } else {
+        this->unk_440 = 0;
+    }
+    Math_SmoothStepToS(&this->actor.shape.rot.z, this->unk_444, 4, arg2, arg2);
+    this->actor.world.rot = this->actor.shape.rot;
+    if (this->actor.shape.rot.z == this->unk_444) {
+        return 1;
+    }
+    return 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/EnSw_Init.s")
+void func_80B0E5E0(EnSw* this, GlobalContext* globalCtx) {
+    s32 pad[2];
+    f32 t;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/EnSw_Destroy.s")
+    if (func_80B0E430(this, 6.0f, 0x3E8, 1, globalCtx)) {
+        t = Rand_ZeroOne();
+        this->unk_444 =
+            ((s16)(20000.0f * t) + 0x2EE0) * (Rand_ZeroOne() >= 0.5f ? 1.0f : -1.0f) + this->actor.world.rot.z;
+        this->unk_388 = Rand_S16Offset(10, 30);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/func_80B0C9F0.s")
+    if (DECR(this->unk_442) == 0) {
+        if (func_80B0DEA8(this, globalCtx, 1)) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_STALWALL_LAUGH);
+            this->unk_442 = 20;
+            this->actionFunc = func_80B0E728;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/func_80B0CBE8.s")
+void func_80B0E728(EnSw* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/func_80B0CCF4.s")
+    if (DECR(this->unk_442) != 0) {
+        if (func_80B0DEA8(this, globalCtx, 1)) {
+            this->unk_448 = player->actor.world.pos;
+            this->unk_448.y += 30.0f;
+            this->unk_444 = func_80B0DE34(this, &this->unk_448);
+            func_80B0E430(this, 6.0f, (u16)0xFA0, 0, globalCtx);
+        } else {
+            this->actionFunc = func_80B0E5E0;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/func_80B0CEA8.s")
+        return;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/func_80B0CF44.s")
+    if (func_80B0DFFC(this, globalCtx) == 0) {
```

Comment:
is this boolean?

Reply:
gonna wait for proper docs before worrying about bools